src/controller/python/chip/bdx/BdxTransfer.py

```diff
@@ -0,0 +1,59 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import asyncio
+from ctypes import c_void_p
+from dataclasses import dataclass
+
+from . import Bdx
+
+
+@dataclass
+class InitMessage:
+    TransferControlFlags: int
+    MaxBlockSize: int
+    StartOffset: int
+    Length: int
+    FileDesignator: bytes
+    Metadata: bytes
+
+
+class BdxTransfer:
+    def __init__(self, bdx_transfer: c_void_p, init_message: InitMessage, data: bytes = None):
+        self.init_message = init_message
+        self._bdx_transfer = bdx_transfer
+        self._data = bytearray(data) if data else None
+
+    async def accept(self):
```

Comment:
Could we type this? And maybe make it `bytes` instead of bytearray and then we need some bytes() conversion